imperfect-perfect.tex

```diff
@@ -433,6 +443,76 @@ \section{Interlude on perfectoid fields}
+a ``quasi-Stein space'' whose global sections are the ring $\bigcap_{r>0} \tilde{\calR}^r_{L,A}$, on which the deck transformations act via the powers of $\varphi$.
+For more on this story, see the aforementioned references such as \cite{fargues-fontaine},
+and also \cite[\S 8.7, 8.8]{kedlaya-liu1} and \cite[\S 4.7]{kedlaya-liu2}.
+\end{remark}
+
+\begin{remark} \label{R:deeply ramified}
+Let $F/K$ be an algebraic extension. In the case where $F = \QQ_p(\mu_{p^\infty})$, we have seen already (Example~\ref{exa:cyclotomic}) that the completion of $F$ is a perfectoid field. 
+This property turns to be closely related to ramification of local fields; let us now recall the precise nature of this relationship.
+
+Coates--Greenberg \cite{coates-greenberg} define $F/K$ to be
+\emph{deeply ramified} if for every finite extension $F'$ of $F$, the trace map
+$\Trace: \frakm_{F'} \to \frakm_F$ is surjective. This holds in particular if $F/K$ is 
+\emph{arithmetically profinite} in the sense of Fontaine--Wintenberger
+\cite{fontaine-wintenberger} (see \cite[Corollary~1.5]{fesenko} for a detailed proof);
+the latter holds in turn if $F/K$ is a not totally unramified Galois extension whose Galois group is a $p$-adic Lie group, by a theorem of Sen \cite{sen-lie}.
+\end{remark}
+
+\begin{lemma} \label{L:deeply ramified is perfectoid}
+Let $F/K$ be an algebraic extension with completion $L$. Then $F/K$ is deeply ramified if and only if $L$ is a perfectoid field.
+\end{lemma}
+\begin{proof}
+Suppose first that $F/K$ is deeply ramified. It is clear that the value group of $L$ must be $p$-divisible (so in particular $L$ is not discretely valued) 
+and the residue field of $L$ must be perfect. To prove that $L$ is perfectoid, it then suffices to choose some $\lambda \in \frakm_F$ such that $\lambda^p$ divides $p$ in $\frako_F$, then check that
+the Frobenius map from $\frako_F/(\lambda)$ to $\frako_F/(\lambda^p)$ is surjective.
+To this end, choose $x \in \frako_F$. Since the set of $p$-th powers in $\frako_F$ is closed,
+the minimum of $\left| x - y^p \right|$ over all $y \in \frako_F$ is achieved by some element $y_0$. 
+If $\left| x - y_0^p \right| > \left| \lambda^p \right|$, 
+we may choose $\mu \in L$ with $\left| \mu^p \right| = \left| x - y_0^p \right|$
+and write the integral closure of $\frako_F$ in $F(x^{1/p})$ as the set of elements of the form
+\[
+\sum_{i=0}^{p-1} z_i \left( \frac{x^{1/p} - y_0}{\mu} \right)^{i} 
+\qquad (z_0,\dots,z_{p-1} \in \frako_F).
+\]
+The trace of such an element is $\sum_{i=0}^{p-1} pz_i (-y_0/\mu)^i$, which is necessarily divisible by $p/\mu^{p-1}$; this contradicts the assumption that $F/K$ is deeply ramified.
+We thus conclude that $L$ is perfectoid.
```

Comment:
The above argument has the form:
```
We want to show that A implies B.  So we suppose A holds, we do some work, and we find that if B failed then A would be false, contradiction, so B is true.
```
Whenever I see this, I always see a hidden contrapositive and want to rewrite it as: 
```
We show that (not B) implies (not A).  So we suppose (not B) holds, we do the same work, and we find that (not A) holds.
```
Which saves the reader from having to unwind at the end, and doesn't require modifying anything else when you are adjacently proving that B implies A to conclude that A iff B.